src/components/mainContainer/tasksContainer/taskContainer/taskContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import './taskContainer.scss'
+import {useEffect, useRef, useState} from 'react'
+import {TaskStatus} from '../../../../customTypings'
+import {Link} from 'react-router-dom'
+import axios from 'axios'
+import {useQueryClient} from 'react-query'
+
+type PropsTaskContainer = {
+    text: string,
+    id: string,
+    taskStatus: TaskStatus,
+    date: string,
+    description: string
+}
+
+const TaskContainer = ({text,id,taskStatus}:PropsTaskContainer) => {
+
+	const queryClient = useQueryClient()
+	const [isOptionsVisible , SetIsOptionsVisible ] = useState(false)
+	const containerReference = useRef<HTMLDivElement>(null)
+
+	useEffect(() => {
+		containerReference.current && setTimeout(() => containerReference.current?.classList.add('animation'),150)
+	}, [containerReference])
+
+	const deleteAnimation = () => {
+		setTimeout(() => containerReference.current?.classList.remove('animation'),100)
+		axios.delete(`/api/tasks/${id}`)

```

Comment:
Przydzą się loadery i obsługa błędów

Reply:
https://www.npmjs.com/package/react-toastify do wyświetlania komunikatów to może się przydać